proparse/src/main/java/org/prorefactor/proparse/antlr4/Lexer.java

```diff
@@ -0,0 +1,1117 @@
+          return id(PreprocessorParser.ANNOTATION);
+        case '[':
+          getChar();
+          return makeToken(PreprocessorParser.LEFTBRACE);
+        case ']':
+          getChar();
+          return makeToken(PreprocessorParser.RIGHTBRACE);
+        case '^':
+          getChar();
+          return makeToken(PreprocessorParser.CARET);
+        case ',':
+          getChar();
+          return makeToken(PreprocessorParser.COMMA);
+        case '!':
+          getChar();
+          return makeToken(PreprocessorParser.EXCLAMATION);
+        case '=':
+          getChar();
+          return makeToken(PreprocessorParser.EQUAL);
+        case '(':
+          getChar();
+          return makeToken(PreprocessorParser.LEFTPAREN);
+        case ')':
+          getChar();
+          return makeToken(PreprocessorParser.RIGHTPAREN);
+        case ';':
+          getChar();
+          return makeToken(PreprocessorParser.SEMI);
+        case '*':
+          getChar();
+          return makeToken(PreprocessorParser.STAR);
+        case '?':
+          getChar();
+          return makeToken(PreprocessorParser.UNKNOWNVALUE);
+        case '`':
+          getChar();
+          return makeToken(PreprocessorParser.BACKTICK);
+
+        case '0':
+        case '1':
+        case '2':
+        case '3':
+        case '4':
+        case '5':
+        case '6':
+        case '7':
+        case '8':
+        case '9':
+          getChar();
+          return digitStart();
+
+        case '.':
+          getChar();
+          return periodStart();
+
+        case '>':
+          getChar();
+          if (currChar == '=') {
+            append();
+            getChar();
+            return makeToken(PreprocessorParser.GTOREQUAL);
+          } else {
+            return makeToken(PreprocessorParser.RIGHTANGLE);
+          }
+
+        case '<':
+          getChar();
+          if (currChar == '>') {
+            append();
+            getChar();
+            return makeToken(PreprocessorParser.GTORLT);
+          } else if (currChar == '=') {
+            append();
+            getChar();
+            return makeToken(PreprocessorParser.LTOREQUAL);
+          } else {
+            return makeToken(PreprocessorParser.LEFTANGLE);
+          }
+
+        case '+':
+          getChar();
+          return plusMinusStart(PreprocessorParser.PLUS);
+        case '-':
+          getChar();
+          return plusMinusStart(PreprocessorParser.MINUS);
+
+        case '#':
+        case '|':
+        case '%':
+          getChar();
+          return id(PreprocessorParser.FILENAME);
+
+        default:
+          if (currInt == EOF_CHAR) {
+            getChar(); // preprocessor will catch any infinite loop on this.
+            return makeToken(PreprocessorParser.EOF, "");
+          } else {
+            getChar();
+            return id(PreprocessorParser.ID);
+          }
+
+      }
+    }
+    } catch(IOException uncaught) {
+      throw new RuntimeException(uncaught);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS00112)